analyses/comparative-RNASeq-analysis/02-thresholds-and-outliers.py

```diff
@@ -0,0 +1,231 @@
+"""
+Generates gene outlier thresholds for normalized samples and uses them to
+calculate the outliers.
+
+Outlier thresholds are calculated according to a modified Tukey method, detailed
+below.
+
+Then, outliers are calculated based on these thresholds. In addition, for each sample,
+the 5% of genes with the highest expression are marked, as well as the genes which
+are excluded by the expression and variance filters from step 1.
+
+The final result is a matrix .rds file of genes by samples where, for each sample,
+each gene may be labeled with zero or more of the following keys (concatenated
+together into a string):
+
+U: Up outlier
+D: Down outlier
+T: in Top five percent of genes for this sample
+F: qc Fail (when less than 5% of genes in the sample have nonzero expression)
+P: gene is droPped by expression or variance filter
+
+
+Modified Tukey Method
+--------------------
+
+The current sample being considered for outlier expression is dropped from the
+dataset, and then the Tukey outlier thresholds of
+first_quartile - (1.5 * IQR) and third_quartile + (1.5 * IQR) are calculated
+for each gene. This allows outlier results to be concordant whether a focus
+sample was originally present in the dataset or whether it is a new sample.
```

Comment:
In terms of what's mathematically necessary, adding or removing a single sample to the dataset does materially affect the thresholds (by a very small magnitude - but with 60k genes there are almost always a few that gain or lose outlier status due to this.) 
If you add 50% more samples, the thresholds would *definitely* need to be recalculated.

The goal of all of this threshold-finagling is to gain any given sample's contribution when focusing on any other sample; but when that sample is the focus, to not include itself in the comparison. 
Because there's a pretty large time gap between [Treehouse compendia](https://treehousegenomics.soe.ucsc.edu/public-data/previous-compendia.html) releases we are often using them to analyze samples that won't be themselves included in the background comparison for months, so when we analyze samples that *are* in the comparison we want to give them the same treatment.

Reply:
>  if you wouldn't mind checking the "Allow edits from maintainers" box

Unfortunately I'm a bit stymied here - that checkbox isn't appearing anywhere in the PR, even after a relog (see screenshot). I'll keep an eye out in case it suddenly appears.

![Screen Shot 2020-02-05 at 3 46 55 PM](https://user-images.githubusercontent.com/19291809/73893672-2636a680-482f-11ea-9b47-59fc92ced4d5.png)

Reply:
Hi @e-t-k, my best guess is that it has something to do with the UCSC-Treehouse permissions because this is coming from an organization’s fork/branch and not from a fork under your individual account.